maintenance/projects/js-toolkit/packages/liferay-npm-bundler/src/dependencies.js

```diff
@@ -57,12 +57,15 @@ export function addPackageDependencies(
 	dependencies = dependencies.concat(extraDependencies);
 
 	const dependencyDirs = dependencies
-		.map((dependency) => {
-			return resolveDependencyDir(basedirPath, packageJson, dependency);
-		})
-		.filter((dependencyDir) => {
-			return dependencyDir != null;
-		});
+		.filter(
+			(dependency) =>
+				!project.presetInfo.isAutopreset ||
+				dependency !== project.presetInfo.name
```

Comment:
Not really, because the filtering is operating in different types of objects (because of the `map()` in the middle).

In any case, filtering several times shouldn't matter too much as long it's one after another, because the conceptual complexity is the same, i.e:

```javascript
let a = '';

for (let i=0; i<100; i++) {
  a += 'x';
}

for (let i=0; i<100; i++) {
  a += 'x';
}
```

is equivalent to:

```javascript
let a = '';

for (let i=0; i<100; i++) {
  a += 'x';
  a += 'x';
}
```

Both in result, execution times, and resource consumption.

There may be a small benefit in doing just one loop because you don't need to allocate the `i` variable twice and because you don't need halve the number of increments to the `i` variable, but those should be insignificant compared to the body of the loop (under normal conditions). Also, the compiler/interpreter should be smart enough to optimize it :crossed_fingers: .

In fact, we have lots of [similar constructs in our codebase](https://github.com/liferay/liferay-frontend-projects/search?q=filter%28Boolean%29).

Reply:
![all over the place](https://media.giphy.com/media/dC8DR1np3MFvVYRazq/giphy.gif)